state/zookeeper/zk.go

```diff
@@ -169,6 +171,33 @@ func (s *StateStore) Get(req *state.GetRequest) (*state.GetResponse, error) {
 	}, nil
 }
 
+// Watch performs a watch operation
+func (s *StateStore) Watch(req *state.GetRequest, handler func(msg *state.GetResponse) error) error {
+	var (
+		events <-chan zk.Event
+		err    error
+	)
+	if _, _, events, err = s.conn.GetW(s.prefixedKey(req.Key)); err != nil {
+		return err
+	}
+	go func() {
+		for e := range events {
+			if req.Key != e.Path {
+				continue
+			}
+			if resp, err := s.Get(req); err != nil {
+				handler(&state.GetResponse{
```

Comment:
Shouldn't this be in some sort of infinite loop to recover in case of error.

Also, this feels a lot like a PubSub - how can the subscriber stop this? For example, what if the client (app) restarts?